src/OpenTelemetry/Metrics/MetricAggregators/GaugeMetricAggregator.cs

```diff
@@ -0,0 +1,115 @@
+// <copyright file="GaugeMetricAggregator.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+
+namespace OpenTelemetry.Metrics
+{
+    internal class GaugeMetricAggregator : IGaugeMetric, IMetricBuilder
+    {
+        private readonly object lockUpdate = new object();
+        private Type valueType;
+        private long longValue;
+        private double doubleValue;
+
+        internal GaugeMetricAggregator(string name, DateTimeOffset startTimeExclusive, KeyValuePair<string, object>[] attributes, bool isDelta)
+        {
+            this.Name = name;
+            this.StartTimeExclusive = startTimeExclusive;
+            this.EndTimeInclusive = startTimeExclusive;
+            this.Attributes = attributes;
+            this.IsDeltaTemporality = isDelta;
```

Comment:
Temporality is not a thing for gauges https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/metrics/v1/metrics.proto#L209-L211